lib/install/inflate-shrinkwrap.js

```diff
@@ -26,7 +26,8 @@ module.exports = function (tree, swdeps, opts, finishInflating) {
   if (!npm.config.get('shrinkwrap') || !npm.config.get('package-lock')) {
     return finishInflating()
   }
-  tree.loaded = true
+  tree.loaded = false
+  tree.hasRequiresFromLock = Object.keys(swdeps).every((d) => swdeps[d].requires)
```

Comment:
At one point during the process of writing this it was possible to produce an install that only had partial resolved information. Buuut…

That doesn't seem to be the case any more. So this can become `some` and we can stop including `requires` on every node.

Relatedly, I'm finding that `npm install xyz` or `npm rm xyz` when you have a legacy package-lock but nothing installed fails to properly upgrade the package-lock, eg:

```
$ node ~/code/npm rm null

added 190 packages and removed 1 package in 6.671s
```